File: app/src/main/java/org/deiverbum/app/data/entity/LHReadingShortEntity.java
package org.deiverbum.app.data.entity;

import static org.deiverbum.app.utils.Constants.LH_READING_SHORT;

import androidx.annotation.NonNull;
import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.Index;
import androidx.room.PrimaryKey;

/**
 * @author A. Cedano
 * @version 1.0
 * @since 2022.2
 */

@Entity(tableName = LH_READING_SHORT,
        indices = {@Index(value = {"text"}, unique = true)}
)
public class LHReadingShortEntity {

    @NonNull
    @PrimaryKey
    @ColumnInfo(name = "readingID")
    public Integer lecturaId=0;

    @NonNull
    @ColumnInfo(name = "text")
    public String texto="";

    @NonNull
    @ColumnInfo(name = "quote")
    public String cita="";


}

File: app/src/main/java/org/deiverbum/app/data/entity/LHReadingShortEntity.kt
package org.deiverbum.app.data.entity

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.Index
import androidx.room.PrimaryKey
import org.deiverbum.app.utils.Constants

/**
 * @author A. Cedano
 * @version 1.0
 * @since 2023.1
 */
@Entity(tableName = Constants.LH_READING_SHORT, indices = [Index(value = ["text"], unique = true)])
class LHReadingShortEntity {
    @JvmField
    @PrimaryKey
    @ColumnInfo(name = "readingID")
    var lecturaId = 0

    @JvmField
    @ColumnInfo(name = "text")
    var texto = ""

    @JvmField
    @ColumnInfo(name = "quote")
    var cita = ""
}